### PR TITLE
parse image pull auth from env

### PR DESCRIPTION
**Pull request by sctb512**
## Details
This commit attempts to avoid loading auth from the nydusd configuration file, which is insecure for users. The auth loaded from env will overwrite the auth loaded from file if it is not null.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/bin/nydusd/main.rs [hunk 1]
```diff
@@ -375,13 +375,18 @@ fn process_fs_service(
     let bootstrap = args.value_of("bootstrap");
     // safe as virtual_mountpoint default to "/"
     let virtual_mnt = args.value_of("virtual-mountpoint").unwrap();
+    // auth for registry
+    let auth = std::env::var("IMAGE_PULL_AUTH").ok();
+
+    info!("[abin] auth: {auth:?}");
```
Comment (jiangliu) [inline src/bin/nydusd/main.rs:381]: Please help to remove this info message, which may leak secret information through log files.

**Comment by jiangliu** [comment]
There's a PR to enable passing nydusd configuration information through pipe instead of a file, seems these two PRs are both to enhance the security of auth info. So what's the relationship between the two PRs?

**Comment by sctb512** [comment]
> There's a PR to enable passing nydusd configuration information through pipe instead of a file, seems these two PRs are both to enhance the security of auth info. So what's the relationship between the two PRs?
> 

The PR https://github.com/dragonflyoss/image-service/pull/1379 attempts to propose an interface for nydus-snapshotter to update configuration (including auth). This PR avoids loading auth from disk when starting `nydusd`. I think they are not in conflict with each other.

In short, we can parse auth from the environment variable and update it via pipe.

File: src/bin/nydusd/main.rs (hunk 1 above)
Comment (sctb512) [reply src/bin/nydusd/main.rs]: Done. This message is for testing purposes only.

File: api/src/config.rs [hunk 2]
```diff
@@ -207,6 +207,19 @@ impl ConfigV2 {
             false
         }
     }
+
+    /// Fill authorization for registry backend.
+    pub fn fill_auth(&self, auth: &Option<String>) -> Self {
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply api/src/config.rs:212]: Rename to `update_auth()`?

[the remark this answers is not in the document]
Comment (sctb512) [reply api/src/config.rs]: Good point. Fixed.

File: api/src/http.rs [hunk 3]
```diff
@@ -31,6 +31,8 @@ pub struct ApiMountCmd {
     pub fs_type: String,
     /// Configuration for the filesystem.
     pub config: String,
+    /// Authorization for private registry.
+    pub auth: Option<String>,
```
Comment (jiangliu) [inline api/src/http.rs:35]: Why is this extension needed? I think callers should merge the auth info into `config`.

Comment (sctb512) [reply api/src/http.rs:35]: Yes, you are correct. Fixed it now.

**Review by jiangliu — APPROVED** [review]

**Review by adam3q — COMMENTED** [review]
Maybe we can prefix the first commit log with nydusd? `nydusd: parse image pull auth from env `

**Comment by sctb512** [comment]
> Maybe we can prefix the first commit log with nydusd? `nydusd: parse image pull auth from env `

Done.

**Review by adam3q — APPROVED** [review]
LGTM, Thanks!

**Review by changweige — COMMENTED** [review]
LGTM, thanks

**Review by changweige — APPROVED** [review]

**Pull request merged**